test/analysis/typeComplexityAnalysis.test.ts

```diff
@@ -269,39 +301,482 @@ describe('Test getQueryTypeComplexity function', () => {
+                                }
+                                ... on Human {
+                                    homePlanet
+                                    friends(first: 3) {
+                                        name
+                                    }
+                                }
+                            }
+                        }`;
+                    // Query 1 + 1 hero + max(Droid 0, Human 3) = 5
+                    mockHumanFriendsFunction.mockReturnValueOnce(3);
+                    expect(getQueryTypeComplexity(parse(query), variables, typeWeights)).toBe(5);
+                });
+
+                test('that contain an object and a non-zero complexity', () => {
+                    query = `
+                        query {
+                            hero(episode: EMPIRE) {
+                                name
+                                friends(first: 3) {
+                                    name
+                                }
+                                ... on Droid {
+                                    primaryFunction
+                                }
+                                ... on Human {
+                                    homePlanet
+                                }
+                            }
+                        }`;
+                    mockCharacterFriendsFunction.mockReturnValueOnce(3);
+                    // Query 1 + 1 hero + 3 friends/character
+                    mockHumanFriendsFunction.mockReturnValueOnce(3);
+                    expect(getQueryTypeComplexity(parse(query), variables, typeWeights)).toBe(5);
+                });
+
+                test('that use a variable', () => {
+                    query = `
+                        query {
+                            hero(episode: EMPIRE) {
+                                name
+                                ... on Droid {
+                                    primaryFunction
+                                    friends(first: $first) {
+                                        name
+                                    }
+                                }
+                                ... on Human {
+                                    homePlanet
+                                }
+                            }
+                        }`;
+                    mockDroidFriendsFunction.mockReturnValueOnce(3);
+                    variables = { first: 3 };
+                    // Query 1 + 1 hero + max(Droid 3, Human 0) = 5
+                    expect(getQueryTypeComplexity(parse(query), variables, typeWeights)).toBe(5);
+                });
+
+                test('that do not have a TypeCondition', () => {
+                    query = `
+                        query {
+                            hero(episode: EMPIRE) {
+                                ... {
+                                    name
+                                    scalarList(first: 1)
```

Comment:
Where is scalar list coming from?

Reply:
OK I see, I don't think you need to create a new type weight object, that's just more work to maintain. It's just hard to follow the nested `describe`s in a file this big using githubs GUI.

Reply:
Amen, It could probably benefit from modularity and splitting into multiple files with a shared `initialize()` method that builds the typeWeight object